src/backend/routers/auth.py

```diff
@@ -1,14 +1,32 @@
+from authlib.integrations.starlette_client import OAuthError
 from fastapi import APIRouter, Depends, HTTPException
 from starlette.requests import Request
+from starlette.responses import RedirectResponse
 
 from backend.config.auth import ENABLED_AUTH_STRATEGY_MAPPING
 from backend.database_models import get_session
 from backend.database_models.database import DBSessionDep
-from backend.schemas.auth import Login
+from backend.schemas.auth import Auth, Login
+from backend.services.auth.utils import (
+    get_or_create_user,
+    is_enabled_authentication_strategy,
+)
 
 router = APIRouter(dependencies=[Depends(get_session)])
 
 
+@router.get("/auth_strategies")
+def get_session():
+    """
+    Retrieves the currently enabled list of Authentication strategies.
+
+
+    Returns:
+        List[str]: List of names for enabled Authentication strategies, can be empty.
+    """
+    return ENABLED_AUTH_STRATEGY_MAPPING.keys()
+
+
 @router.get("/session")
 def get_session(request: Request):
     """
```

Comment:
Maybe it would be better to rename this method too. Also, I'm not completely clear on why we need session middleware for the REST API.